src/components/TestRail/TestRail.js

```diff
@@ -174,15 +182,52 @@ class TestRail {
      *
      * @param {string} runID
      * @param {Result[]} testResults
+     * @param includeCasesOnlyIfExistInRun
      * @returns {Promise<AxiosResponse<*>>}
      */
-    sendBatchResults(runID, testResults) {
+    async sendBatchResults(runID, testResults, includeCasesOnlyIfExistInRun) {
+        if (typeof includeCasesOnlyIfExistInRun !== 'boolean') {
+            includeCasesOnlyIfExistInRun = true; //avoid crashes due to nonexistent cases
+        }
         const url = '/add_results_for_cases/' + runID;
 
         const postData = {
             results: [],
         };
 
+        // Fetch valid case IDs for the given runID
+        let validCaseIds = [];
+        let testsCaseIdPage = [];
+        // Pagination variables
+        let offset = 0;
+        const limit = 250; // Maximum allowed by TestRail
+        let hasMoreResults = true;
+
+        // Fetch valid case IDs from TestRail API
+        try { 
+            while (hasMoreResults) {                
+                const getDataResponse = await this.client.getData(
+                    `/get_tests/${runID}`,
+                    {offset, limit},
+                );
+                testsCaseIdPage = getDataResponse.data.tests.map((test) => test.case_id);
+            
+                if (testsCaseIdPage.length > 0) {
```

Comment:
You can remove this nesting if you do
```suggestion
                if (testsCaseIdPage.length === 0) break;  // we've retrieved all case IDs
```
Then you can `while (true) {` and you no longer need the `hasMoreResults` variable.